### DISCUSSION TITLE
Feature request: option to append the 'model type folder' and the selected subfolder to the Custom 'save images' location.

### DISCUSSION THREAD

**Discussion by hakaserver** [BlafKing/sd-civitai-browser-plus#116] [Ideas]
When using the option to override the download location when saving images, it would be great to also have options to append the model type folder to the path. Just like when downloading the regular model files, it appends the model type (Lora/LyCORIS/Stable-diffusion/etc) to the final download location depending on the type of model.
It would also be great if the subfolder selection worked the same way, to further help organize a local 'database' of the models in a separate base folder, but in the same folder structure.

Example:

> base model location: /stable-diffusion-webui/models
> 
> custom save images location: /stable-diffusion-webui/models_db

Then, when downloading a style lora, we could select a subfolder "Style" and when downloading all the files we would have:

> Save model info: saves to /stable-diffusion-webui/models/Lora/Style/
> 
> Save images: saves to /stable-diffusion-webui/models_db/**_Lora/Style/_**
> 
> Download model: saves to /stable-diffusion-webui/models/Lora/Style/

This would be very beneficial in large collections, in case you needed to quickly revisit a specific model information.

**Comment by BlafKing** [comment]
Sounds good!

I'll add an extra toggle in the settings that you can use to toggle overriding the custom location if a sub folder has been used.

(Since your second example is currently the default behaviour if no custom image location is being used.)

**Reply by hakaserver** [reply]
Thanks! Reading back my post, I might have made myself difficult to understand: the idea is to still use the custom location, but also add the model type folder and the selected sub folder at the end of the path.

Currently, if you type a custom location in the settings and you also select a sub folder when downloading the images, the files will be saved at the `.\custom location\<files here>`. My suggestion is to be able to have the images saved at `.\custom location\model type\sub folder selected\<files here>`, essentialy just changing the base ./models/ folder to the custom location folder set for the images.

Example: if downloading a **LoRA** named **MODEL1** in the subfolder **Favorites**, the model would be saved at:
.\ **models** \Lora\Favorites\MODEL1.safetensors

But the images would use the custom location and be saved at:
.\ **custom location** \Lora\Favorites\MODEL1_.png
The same folder structure, but using the custom location as a base folder.

**Reply by BlafKing** [reply]
I see now! I mis read your first message where you wrote `models_db`, my bad!

I'll add an extra toggle under the custom images path that you can use to enable/disable this functionality! :)

**Comment by BlafKing** [comment]
This has now been added as a toggle in v3.1.0! :)

**Reply by hakaserver** [reply]
Awesome! Thank you so much! ❤️

**Reply by hakaserver** [reply]
I just tested and I think it is not working as intended. I'll open an issue describing it.
